src/normalizer/mod.rs

```diff
@@ -78,17 +72,51 @@ pub trait Normalizer: Sync + Send {
     /// Normalize the provided [`Token`].
     /// Options can be set using the provided [`NormalizerOption`].
     ///
-    /// A Normalizer can return several `Token`s.
-    fn normalize<'o>(
-        &self,
-        token: Token<'o>,
-        options: NormalizerOption,
-    ) -> Box<dyn Iterator<Item = Token<'o>> + 'o>;
+    fn normalize<'o>(&self, mut token: Token<'o>, options: NormalizerOption) -> Token<'o> {
+        if options.create_char_map {
+            match token.char_map.take() {
+                Some(mut char_map) => {
+                    let mut lemma = String::new();
+                    let mut tail = token.lemma.as_ref();
+                    for (_, normalized_len) in char_map.iter_mut() {
+                        let (head, t) = tail.split_at(*normalized_len as usize);
+                        tail = t;
+                        let normalized = self.normalize_str(head);
+                        *normalized_len = normalized.len() as u8;
+                        lemma.push_str(normalized.as_ref());
+                    }
+
+                    token.lemma = Cow::Owned(lemma);
+                    token.char_map = Some(char_map);
+                }
+                None => {
+                    let mut buffer = [0; 4];
+                    let mut char_map = Vec::new();
+                    let mut lemma = String::new();
+                    for c in token.lemma().chars() {
+                        let char_str = c.encode_utf8(&mut buffer);
+                        let normalized = self.normalize_str(char_str);
```

Comment:
so far, performance is not my priority, I want to simplify the introduction of a new normalizer.
In the related issue, I was considering adding a `normalize_char` function in addition to `normalize_str`, then adding a default implementation of `normalize_str` that calls `normalize_char`.
I saw two issues with doing this:
1) We have more methods to implement because calling `normalize_char` in `normalize_str` is not always the best approach. For instance when your normalizer removes characters.
2) All the used libraries don't implement a function that takes a char in parameters

However, you're right we may lose performance due to allocations 🤔
I may find a way to do it by implementing several traits like `CharNormalizer` and `StrNormalizer` with a default implementation of the trait `Normalizer` for each trait, is it at least possible? 🤔